outputs/junit.go

```diff
@@ -40,24 +40,24 @@ func (r JUnit) Output(w io.Writer, results <-chan []resource.TestResult,
 				escapeString(testResult.ResourceId) + " " +
 				testResult.Property + "\" " +
 				"time=\"" + duration + "\">\n"
-			if testResult.Result == resource.FAIL {
+			if testResult.Result == resource.SKIP {
+				summary[testCount] += "<skipped/>"
+				skipped++
+			} else {
+				summary[testCount] += "<system-out>" +
+					escapeString(testResult.Stdout) +
+					"</system-out>\n"
 				summary[testCount] += "<system-err>" +
-					escapeString(humanizeResult2(testResult)) +
+					escapeString(testResult.Stderr) +
 					"</system-err>\n"
-				summary[testCount] += "<failure>" +
-					escapeString(humanizeResult2(testResult)) +
-					"</failure>\n</testcase>\n"
-
-				failed++
-			} else {
-				if testResult.Result == resource.SKIP {
-					summary[testCount] += "<skipped/>"
-					skipped++
+				if testResult.Result == resource.FAIL {
+					summary[testCount] += "<failure>" +
+						escapeString(humanizeResult2(testResult)) +
+						"</failure>\n"
+					failed++
 				}
-				summary[testCount] += "<system-out>" +
-					escapeString(humanizeResult2(testResult)) +
-					"</system-out>\n</testcase>\n"
 			}
+			summary[testCount] += "</testcase>\n"
```

Comment:
Rewrite logic of `junit` outputter: always include `<system-out/>` and `<system-err/>` tags, unless test is skipped.